src/contrib/persistence/Akka.Persistence.Query.Sql/AllEventsPublisher.cs

```diff
@@ -129,6 +157,7 @@ protected bool Replaying( object message )
                     Log.Error(failure.Cause, "event replay failed, due to [{0}]", failure.Cause.Message);
                     Buffer.DeliverBuffer(TotalDemand);
                     OnErrorThenStop(failure.Cause);
+                    QueryPermitter.Tell(ReturnQueryStart.Instance); // return token to permitter
```

Comment:
Return tokens when we fail and when we succeed - always return them _after_ the database has completed the query, successfully or otherwise.